### PR TITLE
Make smoothing optional and fix some pep8 issues

### LINKED ISSUE #15: Make smoothing optional
To increase flexibility for users, it would be nice if smoothing of the lines could be optional.
A checkbox should be added to the GUI with the label "Line smoothing".

### PR DESCRIPTION
**Pull request by havatv**
Make smoothing optional and fix some pep8 issues.

Fix #15

### CODE REVIEW

File: pointtool.py [hunk 1]
```diff
@@ -59,50 +60,49 @@ def snap_tolerance_changed(self, snap_tolerance):
...
-                                                QgsProject.instance())
+                to_coords_provider2 = \
+                get_whole_raster(self.rlayer, QgsProject.instance())
         except PossiblyIndexedImageError:
-            self.iface.messageBar().pushMessage("Missing Layer", 
-                    "Can't trace indexed or gray image", 
-                    level=Qgis.Critical, duration=2)
+            self.iface.messageBar().pushMessage("Missing Layer",
+                            "Can't trace indexed or gray image",
+                   d              level=Qgis.Critical, duration=2)
```
Comment (mkondratyev85) [inline pointtool.py:103]: Please remove 'd' character from the beginning of the line.

**Comment by mkondratyev85** [comment]
I agree with the proposed changes. 

Before I merge this pull request I want to clarify a few things.

1. Please fix typo in pointtool.py on line 113

2. What is the purpose of the combobox right to the checkbox you added?
Right now it will confuse users, i'm afraid.

Снимок экрана от 2020-09-03 10-42-04

File: raster_tracer_dockwidget_base.ui [hunk 2]
```diff
@@ -69,6 +69,16 @@
...
+    <item row="6" column="0">
+     <widget class="QCheckBox" name="checkBoxSmooth">
+      <property name="text">
+       <string>Smooth lines</string>
+      </property>
+     </widget>
+    </item>
+    <item row="6" column="1">
+     <widget class="QComboBox" name="comboBox"/>
+    </item>
```
Comment (mkondratyev85) [inline raster_tracer_dockwidget_base.ui:81]: What is the purpose of this comboBox?

**Comment by havatv** [comment]
> Before I merge this pull request I want to clarify a few things.
>     1. Please fix typo in pointtool.py on line 113

Sorry about that! Fixed now.

>     2. What is the purpose of the combobox right to the checkbox you added?
>        Right now it will confuse users, i'm afraid.

The plan was to let the user choose smoothing method (smoothing / simplification, Douglas-Peucker, Chaiken, Snake, ...). But that can wait. I have removed it.

**Comment by mkondratyev85** [comment]
> The plan was to let the user choose smoothing method (smoothing / simplification, Douglas-Peucker, Chaiken, snakes, ...). But that can wait. I have removed it.


Ok. I wasn't sure if it was intentional or not. Added an issue for that - #17

**Comment by havatv** [comment]
Thanks for handling this so soon, @mkondratyev85.
We are planning to use *Raster Tracer" in course this autumn, so it would be nice to have a new release og the plugin (with this feature included) in the not too distant future... :-)

**Comment by mkondratyev85** [comment]
> We are planning to use *Raster Tracer" in course this autumn, so it would be nice to have a new release og the plugin (with this feature included) in the not too distant future... :-)

Sure. What other futures do you want to see in `raster_tracer` before the release? And what this course will be about?

**Comment by havatv** [comment]
> > We are planning to use *Raster Tracer" in course this autumn, so it would be nice to have a new release og the plugin (with this feature included) in the not too distant future... :-)
> 
> Sure. What other futures do you want to see in `raster_tracer` before the release? And what this course will be about?

Thanks a lot for being so helpful, @mkondratyev85.
I think everything should be fine now. I wanted the plugin to be easy to use for students. The geometry type warning should be helpful. In addition, optional smoothing will help them understand how the tracing is working.

The course is a basic GIS course here at NMBU. Thanks to the *Raster Tracer* plugin, it is now possible for the students to only use QGIS for the exercises.This is particularly useful in these Covid19 times when access to the University computer laboratories is limited. Until last year, they had to use ArcScan for one of the exercises.

Thanks again!

**Pull request merged**